shell/src/main/java/alluxio/shell/command/CopyFromLocalCommand.java

```diff
@@ -80,15 +79,81 @@ public void run(CommandLine cl) throws IOException {
   }
 
   /**
+   * Copies a directory from local to Alluxio filesystem. The destination directory structure
+   * maintained as local directory. This method is used when input path is a directory.
+   *
+   * @param srcDir The source directory in the local filesystem
+   * @param dstPath The {@link AlluxioURI} of the destination
+   * @throws IOException if a non-Alluxio related exception occurs
+   *
+   */
+  private void copyFromLocalDirs(File srcDir, AlluxioURI dstPath) throws IOException {
+    try {
+      boolean existing = mFileSystem.exists(dstPath);
+      createDscDir(dstPath);
+      List<String> errorMessages = Lists.newArrayList();
+      File[] fileList = srcDir.listFiles();
+      int misFiles = 0;
+      for (File srcFile : fileList) {
+        AlluxioURI newURI = new AlluxioURI(dstPath, new AlluxioURI(srcFile.getName()));
+        try {
+          if (srcFile.isDirectory()) {
+            copyFromLocalDirs(srcFile, newURI);
+          } else {
+            copyPath(srcFile, newURI);
+          }
+        } catch (IOException e) {
+          errorMessages.add(e.getMessage());
+          if (!mFileSystem.exists(newURI)) {
+            misFiles++;
+          }
+        }
+      }
+      if (errorMessages.size() != 0) {
+        if (misFiles == fileList.length) {
```

Comment:
@aaudiber If some files copied successful, but only one copied failed. Just as this situation, I think the directory shouldn't be deleted.